accesscontroltool-bundle/src/test/java/biz/netcentric/cq/tools/actool/configreader/YamlMacroProcessorTest.java

```diff
@@ -481,5 +477,75 @@ public void testDefWithUnderscore() throws Exception {
+        AuthorizableConfigBean authorizableConfig3 = groups.getAuthorizableConfig("group-type3-obj1val1");
+        assertNotNull(authorizableConfig3);
+        assertEquals("Type3 obj1val2 obj1val3", authorizableConfig3.getName());
+        assertEquals("obj1val3 => mapped-obj1val3", authorizableConfig3.getDescription());
+        assertNotNull(groups.getAuthorizableConfig("group-type3-obj2val1"));
+        assertNotNull(groups.getAuthorizableConfig("group-type3-obj3val1"));
+
+        AuthorizableConfigBean authorizableConfig4 = groups.getAuthorizableConfig("group-type4-deepListVal1");
+        assertNotNull(authorizableConfig4);
+        assertEquals("Type4 deepListVal1", authorizableConfig4.getName());
+        assertNotNull(groups.getAuthorizableConfig("group-type4-deepListVal2"));
+        assertNotNull(groups.getAuthorizableConfig("group-type4-deepListVal3"));
+
+    }
+
+    @Test
+    public void testDefWithComplexNestedLoops() throws Exception {
+
+
+        List<Map> yamlList = getYamlList("test-def-complex-nested-loops.yaml");
+
+        yamlList = yamlMacroProcessor.processMacros(yamlList, globalVariables, installLog, session);
+
+        AcesConfig aceConfigs = readAceConfigs(yamlList);
+        assertEquals(8, aceConfigs.size());
+        Iterator<AceBean> it = aceConfigs.iterator();
+        
+        String[][] expectedVals = new String[][] {
+                new String[] { "/content/base/level1name1", "<jcr:root jcr:primaryType=\"sling:Folder\" jcr:title=\"level1title1\"/>"},
+                new String[] { "/content/base/level1name1/level2name1", "<jcr:root jcr:primaryType=\"sling:Folder\" jcr:title=\"level2title1\"/>"},
+                new String[] { "/content/base/level1name1/level2name1/level3name1", "<jcr:root jcr:primaryType=\"sling:Folder\" jcr:title=\"level3title1\"/>"},
+                new String[] { "/content/base/level1name1/level2name2", "<jcr:root jcr:primaryType=\"sling:Folder\" jcr:title=\"level2title2\"/>"},
+                new String[] { "/content/base/level1name1/level2name3", "<jcr:root jcr:primaryType=\"sling:Folder\" jcr:title=\"level2title3\"/>"},
+                new String[] { "/content/base/level1name1/level2name3/level3name1", "<jcr:root jcr:primaryType=\"sling:Folder\" jcr:title=\"level3title1\"/>"},
+                new String[] { "/content/base/level1name1/level2name3/level3name2", "<jcr:root jcr:primaryType=\"sling:Folder\" jcr:title=\"level3title2\"/>"},
+                new String[] { "/content/base/level1name1/level2name3/level3name3", "<jcr:root jcr:primaryType=\"sling:Folder\" jcr:title=\"level3title3\"/>"},
+        };
+        
+        for(int i=0; i<expectedVals.length && it.hasNext(); i++) {
+            AceBean aceBean = it.next();
+            assertEquals("Expected path for bean "+i, expectedVals[i][0], aceBean.getJcrPath());
+            assertEquals("Expected initial content for bean "+i, expectedVals[i][1], aceBean.getInitialContent());
+        }
+    
+
+    }
+    
 }
```

Comment:
Can we have a test for global variables as well?

Reply:
Added one in a1feaba